s3/src/main/java/com/yahoo/ycsb/db/S3Client.java

```diff
@@ -0,0 +1,348 @@
+ * The name of the file is determined by the parameter key. 
+ *This key is automatically generated by YCSB.
+ *
+ * @author ivanB1975
+ */
+public class S3Client extends DB {
+ 
+  private static String accessKeyId;
+  private static String secretKey;
+  private static String endPoint;
+  private static String region;
+  private static String maxErrorRetry;
+  private static BasicAWSCredentials s3Credentials;
+  private static AmazonS3Client s3Client;
+  private static ClientConfiguration clientConfig;
+  private static final AtomicInteger INIT_COUNT = new AtomicInteger(0);
+
+  /**
+  * Cleanup any state for this storage.
+  * Called once per S3 instance; 
+  */
+  @Override
+  public void cleanup() throws DBException {
+    if (INIT_COUNT.decrementAndGet() == 0) {
+      try {
+        this.s3Client.shutdown();
+      } catch (Exception e){
+        e.printStackTrace();
+      } finally {
+        this.s3Client = null;
+      }
+    }
+  }
+  /**
+  * Delete a file from S3 Storage.
+  * 
+  * @param bucket
+  *            The name of the bucket
+  * @param key
+  * The record key of the file to delete.
+  * @return Zero on success, a non-zero error code on error. See the
+  * {@link DB} class's description for a discussion of error codes.
+  */
+  @Override
+  public int delete(String bucket, String key) {
+    try {
+      this.s3Client.deleteObject(new DeleteObjectRequest(bucket, key));
+    } catch (Exception e){
+      e.printStackTrace();
+      return 1;
+    }
+    return 0;
+  }
+  /**
+  * Initialize any state for the storage.
+  * Called once per S3 instance; If the client is not null it is re-used.
+  */
+  @Override
+  public void init() throws DBException {
+    INIT_COUNT.incrementAndGet();
+    synchronized (S3Client.class){
+      if (s3Client != null) {
+        System.out.println("Reusing the same client");
+        return;
+      } 
+      try {
+        Properties props = getProperties();
+        accessKeyId = props.getProperty("s3.accessKeyId", "accessKeyId");
+        secretKey = props.getProperty("s3.secretKey", "secretKey");
+        endPoint = props.getProperty("s3.endPoint", "s3.amazonaws.com");
+        region = props.getProperty("s3.region", "us-east-1");
+        maxErrorRetry = props.getProperty("s3.maxErrorRetry", "15");
+        System.out.println("Inizializing the S3 connection");
+        s3Credentials = new BasicAWSCredentials(accessKeyId, secretKey);
+        clientConfig = new ClientConfiguration();
+        clientConfig.setMaxErrorRetry(Integer.parseInt(maxErrorRetry));
+        s3Client = new AmazonS3Client(s3Credentials, clientConfig);
+        s3Client.setRegion(Region.getRegion(Regions.fromName(region)));
+        s3Client.setEndpoint(endPoint);
+        System.out.println("Connection successfully initialized");
+      } catch (Exception e){
+        System.err.println("Could not connect to S3 storage: "+ e.toString());
+        e.printStackTrace();
+        return;
```

Comment:
should probably throw DBException here so that the framework knows to stop rather than attempt to go one with no connection.